internal/auth/repository_auth_method.go

```diff
@@ -0,0 +1,230 @@
+		)
+	}
+
+	return amr.queryAuthMethods(ctx, query, args)
+}
+
+// ListRefresh lists auth methods across all subtypes.
+func (amr *AuthMethodRepository) ListRefresh(ctx context.Context, scopeIds []string, updatedAfter time.Time, afterItem pagination.Item, limit int) ([]AuthMethod, time.Time, error) {
+	const op = "auth.(*AuthMethodRepository).list"
+	switch {
+	case len(scopeIds) == 0:
+		return nil, time.Time{}, errors.New(ctx, errors.InvalidParameter, op, "missing scope ids")
+	case updatedAfter.IsZero():
+		return nil, time.Time{}, errors.New(ctx, errors.InvalidParameter, op, "missing updated after time")
+	case limit < 1:
+		return nil, time.Time{}, errors.New(ctx, errors.InvalidParameter, op, "missing limit")
+	}
+
+	var inClauses []string
+	var args []any
+	for i, scopeId := range scopeIds {
+		arg := "scope_id_" + strconv.Itoa(i)
+		inClauses = append(inClauses, "@"+arg)
+		args = append(args, sql.Named(arg, scopeId))
+	}
+	inClause := strings.Join(inClauses, ", ")
+	whereClause := "scope_id in (" + inClause + ")"
+
+	query := fmt.Sprintf(listAuthMethodsRefreshTemplate, whereClause, limit)
+	args = append(args,
+		sql.Named("updated_after_time", timestamp.New(updatedAfter)),
+	)
+	if afterItem != nil {
+		query = fmt.Sprintf(listAuthMethodsRefreshPageTemplate, whereClause, limit)
+		args = append(args,
+			sql.Named("last_item_update_time", afterItem.GetUpdateTime()),
+			sql.Named("last_item_id", afterItem.GetPublicId()),
+		)
+	}
+
+	return amr.queryAuthMethods(ctx, query, args)
+}
+
+// EstimatedCount estimates the total number of auth methods.
+func (amr *AuthMethodRepository) EstimatedCount(ctx context.Context) (int, error) {
+	const op = "auth.(*AuthMethodRepository).EstimatedCount"
+	rows, err := amr.reader.Query(ctx, estimateCountAuthMethodsQuery, nil)
+	if err != nil {
+		return 0, errors.Wrap(ctx, err, op, errors.WithMsg("failed to query total auth methods"))
+	}
+	var count int
+	for rows.Next() {
+		if err := amr.reader.ScanRows(ctx, rows, &count); err != nil {
+			return 0, errors.Wrap(ctx, err, op, errors.WithMsg("failed to query total auth methods"))
+		}
+	}
+	return count, nil
+}
+
+// ListDeletedIds lists the deleted IDs of all auth methods since the time specified.
+func (amr *AuthMethodRepository) ListDeletedIds(ctx context.Context, since time.Time) ([]string, time.Time, error) {
+	const op = "auth.(*AuthMethodRepository).ListDeletedIds"
+	var deletedAuthMethodIDs []string
+	var transactionTimestamp time.Time
+	if _, err := amr.writer.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{}, func(r db.Reader, w db.Writer) error {
+		rows, err := amr.writer.Query(ctx, listDeletedIdsQuery, []any{sql.Named("since", since)})
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		defer rows.Close()
+		for rows.Next() {
+			var id string
+			if err := rows.Scan(&id); err != nil {
+				return errors.Wrap(ctx, err, op)
+			}
+			deletedAuthMethodIDs = append(deletedAuthMethodIDs, id)
+		}
```

Comment:
Can't we scan directly into `deletedAuthMethodIDs`?
```suggestion
		for rows.Next() {
			if err := r.ScanRows(ctx, &deletedAuthMethodIDs); err != nil {
				return errors.Wrap(ctx, err, op)
			}
		}
```